### ISSUE TITLE
Dropdown is not working properly when contains a dropdown inside

### ISSUE THREAD

**Issue by azahreba** [angular-ui/bootstrap#3957]
I guess a child dropdown fires an event or smth that affects the parent dropdown
http://plnkr.co/edit/Ddfgj4JGvhPj2cGwHC5o?p=preview

**Comment by wesleycho** [comment]
Currently, the child dropdown will not toggle - you can observe this [here](http://plnkr.co/edit/EnTRwBS7UxHDfgDacdss?p=preview).

**Comment by azahreba** [comment]
I tried to investigate this issue and the first reason is ```DropdownController: scope.isOpen```, after assigning this flag to other property inside the scope or to the controller itself - it works.
Then I had to refactor ```dropdownService``` with the new property, but it breaks auto-closing with "outsideClick " because of the accumulated ```dropdownService: openScope```.
So that there was a situation when I would have to close child and the parent dropdowns after the click outside them - in this case it's completely different logic for the service.(((

**Comment by azahreba** [comment]
I fixed the issue. Will prepare a commit.

**Comment by wesleycho** [comment]
As noted in #2421, we will not support this due to the complexity this adds to supporting existing features & the amount of breaking code this will likely result in.

**Comment by azahreba** [comment]
Actually, I fixed it in 3 lines in JS, without any tricks with CSS or HTML.
https://github.com/azahreba/bootstrap/commit/b86cbda99275670e27aa94f293423ee7cef303d7 (a simple factory pattern is used there).
Now there are 2 failed unit-tests and that's the question, how to fix it.
Because I've tried to reproduce those unit tests on the real app, and everything is working properly with ```class="open"```.
